src/WorkflowCore/Models/LifeCycleEvents/WorkflowTerminated.cs

```diff
@@ -1,10 +1,9 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
 
 namespace WorkflowCore.Models.LifeCycleEvents
 {
     public class WorkflowTerminated : LifeCycleEvent
     {
+        public Exception Exception { get; set; }
```

Comment:
This class also needs to be serializable

Reply:
Would it be fine if we would put `ExecutionError` in here instead?

Reply:
There already is ` WorkflowError` lifecycle event.  Also, you could terminate a workflow without an error... not sure this field makes sense?

Reply:
@danielgerlag 
When a workflow doesn't catch an exception thrown in `Try` clause, it seems logical to me that the workflow has to be terminated, because that means something has gone pretty badly. This can be the case if there is no matching exception type in corresponding `Catch` clauses. It means `WorkflowTerminated` event has to be published when this happens. I publish `WorkflowError` event when an exception happens inside of a `Try` clause, please, correct me, if I'm doing it wrong. If we want to retrieve information about why workflow has been terminated, we have to preserve an exception in some way. If a workflow was terminated without an error, this field just equals to `null`. 
Nevertheless, if this field needs to be serializable, I'm going to make it of a custom class with fields like Type, Message and StackTrace, because I feel like people might make a good use of the additional information on an unexpected workflow termination.